tests/vcs/mercurial/HgCommitBearTest.py

```diff
@@ -0,0 +1,504 @@
+
+    @staticmethod
+    def run_hg_command(*args, stdin=None):
+        run_shell_command(' '.join(('hg',) + args), stdin)
+
+    @staticmethod
+    def hg_commit(msg):
+        if(msg == ''):
+            msg = ' '
+        with open(HgCommitBearTest.testFile, 'w') as f:
+            f.write(msg)
+        HgCommitBearTest.run_hg_command('commit -l ',
+                                        HgCommitBearTest.testFile)
+
+    def changeHgConfig(self, newConfig):
+        with open(self.hgdir+'/.hg/hgrc', 'w') as f:
+            f.write(newConfig)
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        return list(result.message for result in self.uut.run(*args, **kwargs))
+
+    def assert_no_msgs(self):
+        """
+        Assert that there are no messages in the message queue of the bear, and
+        show the messages in the failure message if it is not empty.
+        """
+        self.assertTrue(
+            self.msg_queue.empty(),
+            'Expected no messages in bear message queue, but got: ' +
+            str(list(str(i) for i in self.msg_queue.queue)))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = HgCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.hgdir = mkdtemp()
+        os.chdir(self.hgdir)
+        self.run_hg_command('init')
+        run_shell_command('touch '+HgCommitBearTest.testFile)
+        self.run_hg_command('add', HgCommitBearTest.testFile)
+        newConfigFile = ('[ui]\n'
+                         'username = user.email <coala@coala.io>')
+        self.changeHgConfig(newConfigFile)
+        self.run_hg_command('commit -m "testInit Commit"')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.hgdir, onerror=onerror)
+
+    def test_check_prerequisites(self):
+        _shutil_which = shutil.which
+        try:
+            shutil.which = lambda *args, **kwargs: None
+            self.assertEqual(HgCommitBear.check_prerequisites(),
+                             'hg is not installed.')
+
+            shutil.which = lambda *args, **kwargs: 'path/to/hg'
+            self.assertTrue(HgCommitBear.check_prerequisites())
+        finally:
+            shutil.which = _shutil_which
+
+    def test_get_metadata(self):
```

Comment:
this belongs in new `CommitBearTest` module&class, as it doesnt depend on git vs hg